src/IROperator.cpp

```diff
@@ -1497,7 +1497,7 @@ Expr saturating_cast(Type t, Expr e) {
 Expr select(Expr condition, Expr true_value, Expr false_value) {
     if (as_const_int(condition)) {
         // Why are you doing this? We'll preserve the select node until constant folding for you.
```

Comment:
What's the problem with immediately throwing it away depending on the value of the constant?

Reply:
At the time I wrote the original code, I wasn't sure in what circumstances that would ever happen so I was hesistant to not return a Select node. Maybe you're right though. It would probably only be used that way in generic code, where something is sometimes an Expr and sometimes a constant, for confused people who think you can't use a ternary operator in that case to switch between expressions based on a bool, or for people who prefer the select syntax over C++'s or python's ternary operator for some reason. In all of those cases it's fine to eagerly simplify.